test/units/galaxy/test_api.py

```diff
@@ -910,3 +927,125 @@ def test_get_role_versions_pagination(monkeypatch, responses):
+    GalaxyAPI(None, "test", 'https://galaxy.ansible.com/', no_cache=False)
+
+    assert os.path.isdir(cache_dir)
+    with open(cache_file) as fd:
+        actual_cache = fd.read()
+    assert actual_cache == cache_file_contents
+    assert stat.S_IMODE(os.stat(cache_file).st_mode) == 0o655
+
+
+@pytest.mark.parametrize('content', [
+    '',
+    'value',
+    '{"de" "finit" "ely" [\'invalid"]}',
+    '[]',
+    '{"version": 2, "test": "json"}',
+    '{"version": 2, "key": "ÅÑŚÌβŁÈ"}',
+])
+def test_cache_invalid_cache_content(content, cache_dir):
+    cache_file = os.path.join(cache_dir, 'api.json')
+    with open(cache_file, mode='w') as fd:
+        fd.write(content)
+        os.chmod(cache_file, 0o664)
+
+    GalaxyAPI(None, "test", 'https://galaxy.ansible.com/', no_cache=False)
+
+    with open(cache_file) as fd:
+        actual_cache = fd.read()
+    assert actual_cache == '{"version": 1}'
+    assert stat.S_IMODE(os.stat(cache_file).st_mode) == 0o664
+
+
+def test_world_writable_cache(cache_dir, monkeypatch):
+    mock_warning = MagicMock()
+    monkeypatch.setattr(Display, 'warning', mock_warning)
+
+    cache_file = os.path.join(cache_dir, 'api.json')
+    with open(cache_file, mode='w') as fd:
+        fd.write('{"version": 2}')
+        os.chmod(cache_file, 0o666)
+
+    api = GalaxyAPI(None, "test", 'https://galaxy.ansible.com/', no_cache=False)
+    assert api._cache is None
+
+    with open(cache_file) as fd:
+        actual_cache = fd.read()
+    assert actual_cache == '{"version": 2}'
+    assert stat.S_IMODE(os.stat(cache_file).st_mode) == 0o666
+
+    assert mock_warning.call_count == 1
+    assert mock_warning.call_args[0][0] == \
+        'Galaxy cache has world writable access (%s), ignoring it as a cache source.' % cache_file
+
+
+def test_no_cache(cache_dir):
+    cache_file = os.path.join(cache_dir, 'api.json')
+    with open(cache_file, mode='w') as fd:
+        fd.write('random')
+
+    api = GalaxyAPI(None, "test", 'https://galaxy.ansible.com/')
+    assert api._cache is None
+
+    with open(cache_file) as fd:
+        actual_cache = fd.read()
+    assert actual_cache == 'random'
+
+
+def test_clear_cache_with_no_cache(cache_dir):
+    cache_file = os.path.join(cache_dir, 'api.json')
+    with open(cache_file, mode='w') as fd:
+        fd.write('{"version": 1, "key": "value"}')
+
+    GalaxyAPI(None, "test", 'https://galaxy.ansible.com/', clear_response_cache=True)
+    assert not os.path.exists(cache_file)
+
+
+def test_clear_cache(cache_dir):
+    cache_file = os.path.join(cache_dir, 'api.json')
+    with open(cache_file, mode='w') as fd:
+        fd.write('{"version": 1, "key": "value"}')
+
+    GalaxyAPI(None, "test", 'https://galaxy.ansible.com/', clear_response_cache=True, no_cache=False)
+
+    with open(cache_file) as fd:
+        actual_cache = fd.read()
+    assert actual_cache == '{"version": 1}'
+    assert stat.S_IMODE(os.stat(cache_file).st_mode) == 0o600
+
+
+@pytest.mark.parametrize('url, expected', [
```

Comment:
I learned recently that you can use actual separate literals to improve readability:
```suggestion
@pytest.mark.parametrize(['url', 'expected'], [
```